csp_collector.go

```diff
@@ -36,33 +37,96 @@ var (
 
 	// Flag for toggling verbose output.
 	debugFlag bool
+
+	// Path to file which has blocked URI's per line
+	blockedURIfile string
+
+	// Default URI Filter list
+	ignoredBlockedURIs = []string{
+		"resource://",
+		"chromenull://",
+		"chrome-extension://",
+		"safari-extension://",
+		"mxjscall://",
+		"webviewprogressproxy://",
+		"res://",
+		"mx://",
+		"safari-resource://",
+		"chromeinvoke://",
+		"chromeinvokeimmediate://",
+		"mbinit://",
+		"opera://",
+		"localhost",
+		"127.0.0.1",
+		"none://",
+		"about:blank",
+		"android-webview",
+		"ms-browser-extension",
+		"wvjbscheme://__wvjb_queue_message__",
+		"nativebaiduhd://adblock",
+		"bdvideo://error",
+	}
+
+	// TCP Port to listen on
+	listenPort int
 )
 
 func setupDebugLogger(debugHandle io.Writer) {
 	debug = log.New(debugHandle, "[DEBUG] ", log.Lmicroseconds)
 }
 
+func trimEmpty (s []string) []string {
+    var r []string
+    for _, str := range s {
+        if str != "" {
+            r = append(r, str)
+        }
+    }
+    return r
+}
+
 func main() {
 	setupDebugLogger(os.Stdout)
 
 	version := flag.Bool("version", false, "Display the version")
 	flag.BoolVar(&debugFlag, "debug", false, "Output additional logging for debugging")
+	flag.StringVar(&blockedURIfile, "filter-file", "", "Blocked URI Filter file")
+	flag.IntVar(&listenPort, "port", 8080, "Port to listen on")
 
 	flag.Parse()
 
 	if *version {
 		fmt.Printf("csp-collector (%s)\n", Rev)
-		os.Exit(0)
```

Comment:
What is the thinking here? The intention is that if you run the binary with `--version` that is all you are expecting to get back. Without this `os.Exit`, execution continues.

Reply:
Removed by mistake, restoring